comitup/statemgr.py

```diff
@@ -115,20 +116,31 @@ def get_info(conf, data):
     return info
 
 
-def expand_ap(ap_name, id):
+def expand_ap(ap_name, data):
     returnval = ap_name
+    expand_spec_srch = re.search(REGEX_APNAME_SPEC, ap_name)
+    if expand_spec_srch is not None:
+        # There is a'-<###>' section of the ap_name, process it
+        expand_spec = expand_spec_srch.group()
+        num = len(expand_spec)
 
-    for num in range(5):
```

Comment:
I prefer this mechanism for doing the replacements. With it, there is no need for the restriction on the SSID format. Just add additional loops for the MAC and serial.

Reply:
I can follow the previous logic.  However having the regex check the length of `n, M, s` was also easy to do...

Reply:
I noticed that you are only looking at the first character of the e.g. MMM string.

Reply:
Right, by the regex rule validation, all other letters are known to be the same.